pimd/pim_nb_config.c

```diff
@@ -256,19 +257,18 @@ static int pim_ssm_cmd_worker(struct pim_instance *pim, const char *plist,
 	return ret;
 }
 
-static int pim_rp_cmd_worker(struct pim_instance *pim,
-		struct in_addr rp_addr,
-		struct prefix group, const char *plist,
-		char *errmsg, size_t errmsg_len)
+static int pim_rp_cmd_worker(struct pim_instance *pim, pim_addr rp_addr,
+			     struct prefix group, const char *plist,
+			     char *errmsg, size_t errmsg_len)
 {
-	char rp_str[INET_ADDRSTRLEN];
 	int result;
 
-	inet_ntop(AF_INET, &rp_addr, rp_str, sizeof(rp_str));
-
 	result = pim_rp_new(pim, rp_addr, group, plist, RP_SRC_STATIC);
 
 	if (result == PIM_RP_NO_PATH) {
+		char rp_str[PIM_ADDRSTRLEN];
+
+		snprintfrr(rp_str, sizeof(rp_str), "%pPA", &rp_addr);
 		snprintf(errmsg, errmsg_len,
 			 "No Path to RP address specified: %s", rp_str);
```

Comment:
No point in `snprintfrr` + `snprintf`, just use `snprintfrr` directly to `errmsg`

Reply:
Fixed